### PR TITLE
CORE-17327 - workermonitor-refactor

### PR DESCRIPTION
**Pull request by driessamyn**
Remove `WorkerMonitor` so it is no longer tangled up with the WebServer lifecycle. Instead each worker has as WebServer and is configured to use it for metrics and health.

### CODE REVIEW

File: git-hours.sh [hunk 1]
```diff
@@ -0,0 +1,36 @@
+#! /bin/sh
```
Comment (vkolomeyko) [inline git-hours.sh]: These 3 files probably should not be included in this PR.

File: applications/workers/worker-common/src/test/resources/scratch_2.kts [hunk 2]
```diff
@@ -0,0 +1,77 @@
+
```
Comment (vkolomeyko) [inline applications/workers/worker-common/src/test/resources/scratch_2.kts]: Do we need this file in this PR?

File: git-hours.sh (hunk 1 above)
Comment (driessamyn) [reply git-hours.sh]: Sorry, not sure how I missed that in the self-review 🤔 
Removed now, thanks!

File: applications/workers/release/p2p-gateway-worker/addDependencies.txt [hunk 3]
```diff
@@ -0,0 +1,2321 @@
+> Task :buildSrc:extractPluginRequests UP-TO-DATE
```
Comment (jujoramos) [inline applications/workers/release/p2p-gateway-worker/addDependencies.txt]: Guess this file shouldn't be checked in?.

File: applications/workers/worker-common/src/main/kotlin/net/corda/applications/workers/workercommon/Health.kt [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+import net.corda.rest.ResponseCode
+import net.corda.web.api.Endpoint
+import net.corda.web.api.HTTPMethod
+import net.corda.web.api.WebHandler
+import net.corda.web.api.WebServer
+import org.slf4j.LoggerFactory
+import java.util.concurrent.ConcurrentHashMap
+
+object Health {
+    private val logger = LoggerFactory.getLogger(Metrics::class.java)
```
Comment (jujoramos) [inline applications/workers/worker-common/src/main/kotlin/net/corda/applications/workers/workercommon/Health.kt]: I think you want the logger for `Health` here, not `Metrics`.

**Review by jujoramos — APPROVED** [review]
LGTM

**Review by ronanbrowne — APPROVED** [review]
LGTM - blt files

**Review by JamesHR3 — APPROVED** [review]

**Review by charlieR3 — APPROVED** [review]
LGTM - network team review

**Review by emilybowe — APPROVED** [review]
LGTM

File: applications/workers/release/combined-worker/build.gradle [hunk 5]
```diff
@@ -68,6 +68,7 @@ dependencies {
     implementation project(':libs:crypto:crypto-config-impl')
     implementation project(':libs:crypto:crypto-core')
     implementation project(':libs:crypto:crypto-impl')
+    implementation project(":libs:lifecycle:registry")
```
Comment (vkolomeyko) [inline applications/workers/release/combined-worker/build.gradle:71]: Super-nit:
```suggestion
    implementation project(':libs:lifecycle:registry')
```

**Review by vkolomeyko — APPROVED** [review]
LGTM

File: applications/workers/release/combined-worker/build.gradle (hunk 5 above)
Comment (driessamyn) [reply applications/workers/release/combined-worker/build.gradle:71]: Good spot. will fix that in the next PR if that's ok, to avoid another set of approvals.
